File: 202-happy-number/202-happy-number.cs
public class Solution {
    public bool IsHappy(int n) {
        int count = 0;
        while (n > 1)
        {
            int sum = 0;
            int n2 = n;
            while (n2 >= 1)
            {
                sum += (int)Math.Pow(n2 % 10, 2);
                n2 /= 10;
            }
            n = sum;
            count++;
            if (count >= 20) return false;
        }
        return true;
    }
}

File: 202-happy-number/202-happy-number.cpp
class Solution {
public:
    int squareOfSum(int n)  {
        int sum{0}, digit;
        while   (n) {
            digit = n % 10;
            sum += digit * digit;
            n /= 10;
        }
        return sum;
    }
    bool isHappy(int n) {
        int fast , slow;
        fast = slow = n;
        do  {
            slow = squareOfSum(slow);
            fast = squareOfSum(fast);
            fast = squareOfSum(fast);
            if(fast == 1) return 1;
        } while (fast != slow);
        return 0;
    }
};